packages/hardhat-core/src/internal/core/providers/gas-providers.ts

```diff
@@ -135,36 +137,135 @@ export class AutomaticGasProvider extends MultipliedGasEstimationProvider {
 }
 
 export class AutomaticGasPriceProvider extends ProviderWrapper {
-  private _gasPrice: string | undefined;
+  // The amount of blocks you are expected to wait when using EIP-1559.
+  public static readonly EIP1559_BASE_FEE_BLOCKS_PREFERENCE: number = 3;
+
+  // See eth_feeHistory for an explanation of what this means
+  public static readonly EIP1559_REWARD_PERCENTILE: number = 0.5;
+
+  private _nodeHasFeeHistory?: boolean;
 
   public async request(args: RequestArguments): Promise<unknown> {
-    if (args.method === "eth_sendTransaction") {
-      const params = this._getParams(args);
+    if (args.method !== "eth_sendTransaction") {
+      return this._wrappedProvider.request(args);
+    }
 
-      // TODO: Should we validate this type?
-      const tx = params[0];
-      // temporary change to ignore EIP-1559
-      if (
-        tx !== undefined &&
-        tx.gasPrice === undefined &&
-        tx.maxFeePerGas === undefined &&
-        tx.maxPriorityFeePerGas === undefined
-      ) {
-        tx.gasPrice = await this._getGasPriceAsQuantity();
-      }
+    const params = this._getParams(args);
+
+    // TODO: Should we validate this type?
+    const tx = params[0];
+
+    if (tx === undefined) {
+      return this._wrappedProvider.request(args);
+    }
+
+    // We don't need to do anything in these cases
+    if (
+      tx.gasPrice !== undefined ||
+      (tx.maxFeePerGas !== undefined && tx.maxPriorityFeePerGas !== undefined)
```

Comment:
I don't quite understand how the fact that this is a `&&` and not a `||` affects the rest of the code. I mean, what happens when only one of these is defined? For example, later we do:

```
    if (eip1559Values === undefined) {
      const gasPrice = await this._getGasPrice();

      eip1559Values = {
        maxFeePerGas: gasPrice,
        maxPriorityFeePerGas: gasPrice,
      };
    }
```

Does that mean that one of the values set by the user could be overridden?

Reply:
This if handles the case where everything was provided by the user.

The one in your snippet is used if `eth_feeHistory` is not available. I renamed  `eip1559Values` to avoid this confusion.